test/test_um2netcdf.py

```diff
@@ -1017,67 +974,52 @@ def test_fix_cell_methods_keep_weeks():
-            if custom_coord:
-                level_coords.update(custom_coord)
+def level_coords_cube(level_coords):
+    return DummyCube(0, coords=level_coords)
 
-        def coord(self, key):
-            if key in level_coords:
-                return level_coords[key]
 
-            msg = f"{self.__class__}: lacks coord for '{key}'"
-            raise iris.exceptions.CoordinateNotFoundError(msg)
-
-    # return class for instantiation in tests
-    return FakeCubeCoords
-
-
-def test_fix_level_coord_modify_cube_with_rho(level_heights,
-                                              get_fake_cube_coords,
+def test_fix_level_coord_modify_cube_with_rho(level_coords_cube,
+                                              level_heights,
                                               z_sea_rho_data,
                                               z_sea_theta_data):
     # verify cube renaming with appropriate z_rho data
-    cube = get_fake_cube_coords()
-    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name is None
-    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name is None
-    assert cube.coord(um2nc.SIGMA).var_name is None
+    cube = level_coords_cube
+
+    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name != um2nc.MODEL_RHO_LEVEL
+    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name != um2nc.RHO_LEVEL_HEIGHT
+    assert cube.coord(um2nc.SIGMA).var_name != um2nc.SIGMA_RHO
 
     rho = np.ones(z_sea_theta_data.shape) * level_heights[0]
     um2nc.fix_level_coord(cube, rho, z_sea_theta_data)
 
-    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name == "model_rho_level_number"
-    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name == "rho_level_height"
-    assert cube.coord(um2nc.SIGMA).var_name == "sigma_rho"
+    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name == um2nc.MODEL_RHO_LEVEL
+    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name == um2nc.RHO_LEVEL_HEIGHT
+    assert cube.coord(um2nc.SIGMA).var_name == um2nc.SIGMA_RHO
 
 
 def test_fix_level_coord_modify_cube_with_theta(level_heights,
-                                                get_fake_cube_coords,
+                                                level_coords_cube,
                                                 z_sea_rho_data,
                                                 z_sea_theta_data):
     # verify cube renaming with appropriate z_theta data
-    cube = get_fake_cube_coords()
+    cube = level_coords_cube
     um2nc.fix_level_coord(cube, z_sea_rho_data, z_sea_theta_data)
 
-    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name == "model_theta_level_number"
-    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name == "theta_level_height"
-    assert cube.coord(um2nc.SIGMA).var_name == "sigma_theta"
+    assert cube.coord(um2nc.MODEL_LEVEL_NUM).var_name == um2nc.MODEL_THETA_LEVEL_NUM
+    assert cube.coord(um2nc.LEVEL_HEIGHT).var_name == um2nc.THETA_LEVEL_HEIGHT
+    assert cube.coord(um2nc.SIGMA).var_name == um2nc.SIGMA_THETA
 
```

Comment:
There seen to be some subtle differences between iris cubes and our `DummyCube` which would cause the  `test_fix_level_coord_...` tests to fail if we were using real iris cubes.

The `cube.coord()` method uses the current name for a `DimCoord` as its argument, where the "current name" comes from either the `DimCoord`'s `standard_name`, `long_name`,  or `var_name`, with the following order of priority: `standard_name` > `long_name` > `var_name`.

This means that if a `DimCoord` only has a `var_name` attribute, changing the `var_name` forces us to also change the argument in `cube.coord(...)` used to access it:

A made up example after modifying a loaded cube's coordinate names:
```Python 
>>> cube.coord("pseudo_level").__dict__
{'_metadata_manager': MetadataManager(standard_name=None, long_name=None, var_name='pseudo_level', units=Unit('1'), attributes={}, coord_system=None, climatological=False, circular=False), '_values_dm': DataManager(array([1, 2, 3, 4, 5, 6])), '_bounds_dm': None, '_ignore_axis': False}

>>> cube.coord("pseudo_level").var_name = "new_var_name"

>>> cube.coord("pseudo_level")

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/g/data/vk83/apps/payu/1.1.5/lib/python3.10/site-packages/iris/cube.py", line 2304, in coord
    raise iris.exceptions.CoordinateNotFoundError(emsg)
iris.exceptions.CoordinateNotFoundError: "Expected to find exactly 1 'pseudo_level' coordinate, but found none."

>>> cube.coord("new_var_name")
<DimCoord: new_var_name / (1)  [1, 2, ..., 5, 6]  shape(6,)>
```
In these tests however, we're changing the coordinate's `var_names` but not the `cube.coord(...)` arguments used to access them.

Given that the tests are working and it's a fairly subtly difference, should we defer this to a separate issue?

Reply:
Yep, let's defer this to a "cleanup issue" to help focus on finishing off modularisation & the test retrofit.

I'll sort out the new issue...